<a href="https://colab.research.google.com/github/hannahcanon09/project2/blob/main/Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**PROJECT 2**

Pick 2 datasets and answer the following questions:

1) Source of data

2) Brief description of data

3) What is the target?

4) Is this a classification or regression problem?

5) How many features?

6) How many rows of data.

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?

In [ ]:
# mount drive
from google.colab import drive
drive.mount ('/content/drive')

Mounted at /content/drive


In [ ]:
# import
import pandas as pd

# Dataset 1: Predicting Customer Churn

1) Source of the data: Kaggle
[Ecommerce Customer Churn Analysis and Prediction](https://www.kaggle.com/datasets/ankitverma2010/ecommerce-customer-churn-analysis-and-prediction)

2) Brief Description of data
> The data includes characteristics of e-commerce customers. Specifically, it includes their churn or if customers are likely to make another purchase from the business.

In [ ]:
file1 = '/content/drive/MyDrive/Colab Notebooks/Data /E Commerce Dataset.xlsx'

# Read the Excel file into a dictionary of DataFrames
data_dict = pd.read_excel(file1,sheet_name='Data Dict', header = 1)
data_dict

Unnamed: 0    Data                     Variable  \
0          NaN  E Comm                   CustomerID   
1          NaN  E Comm                        Churn   
2          NaN  E Comm                       Tenure   
3          NaN  E Comm         PreferredLoginDevice   
4          NaN  E Comm                     CityTier   
5          NaN  E Comm              WarehouseToHome   
6          NaN  E Comm         PreferredPaymentMode   
7          NaN  E Comm                       Gender   
8          NaN  E Comm               HourSpendOnApp   
9          NaN  E Comm     NumberOfDeviceRegistered   
10         NaN  E Comm             PreferedOrderCat   
11         NaN  E Comm            SatisfactionScore   
12         NaN  E Comm                MaritalStatus   
13         NaN  E Comm              NumberOfAddress   
14         NaN  E Comm                     Complain   
15         NaN  E Comm  OrderAmountHikeFromlastYear   
16         NaN  E Comm                   CouponUsed   
17         NaN  E Comm                   OrderCount   
18         NaN  E Comm            DaySinceLastOrder   
19         NaN  E Comm               CashbackAmount   

                                          Discerption  
0                                  Unique customer ID  
1                                          Churn Flag  
2                  Tenure of customer in organization  
3                  Preferred login device of customer  
4                                           City tier  
5   Distance in between warehouse to home of customer  
6                Preferred payment method of customer  
7                                  Gender of customer  
8   Number of hours spend on mobile application or...  
9   Total number of deceives is registered on part...  
10  Preferred order category of customer in last m...  
11          Satisfactory score of customer on service  
12                         Marital status of customer  
13  Total number of added added on particular cust...  
14        Any complaint has been raised in last month  
15       Percentage increases in order from last year  
16  Total number of coupon has been used in last m...  
17  Total number of orders has been places in last...  
18                   Day Since last order by customer  
19                     Average cashback in last month

In [ ]:
df1 = pd.read_excel(file1, sheet_name='E Comm')
df1.head()

CustomerID  Churn  Tenure PreferredLoginDevice  CityTier  WarehouseToHome  \
0       50001      1     4.0         Mobile Phone         3              6.0   
1       50002      1     NaN                Phone         1              8.0   
2       50003      1     NaN                Phone         1             30.0   
3       50004      1     0.0                Phone         3             15.0   
4       50005      1     0.0                Phone         1             12.0   

  PreferredPaymentMode  Gender  HourSpendOnApp  NumberOfDeviceRegistered  \
0           Debit Card  Female             3.0                         3   
1                  UPI    Male             3.0                         4   
2           Debit Card    Male             2.0                         4   
3           Debit Card    Male             2.0                         4   
4                   CC    Male             NaN                         3   

     PreferedOrderCat  SatisfactionScore MaritalStatus  NumberOfAddress  \
0  Laptop & Accessory                  2        Single                9   
1              Mobile                  3        Single                7   
2              Mobile                  3        Single                6   
3  Laptop & Accessory                  5        Single                8   
4              Mobile                  5        Single                3   

   Complain  OrderAmountHikeFromlastYear  CouponUsed  OrderCount  \
0         1                         11.0         1.0         1.0   
1         1                         15.0         0.0         1.0   
2         1                         14.0         0.0         1.0   
3         0                         23.0         0.0         1.0   
4         0                         11.0         1.0         1.0   

   DaySinceLastOrder  CashbackAmount  
0                5.0          159.93  
1                0.0          120.90  
2                3.0          120.28  
3                3.0          134.07  
4                3.0          129.60

3) What is the target?
> The target is to determine if given a set of characteristics, will a customer make another purchase from the e-commerce platform.

4) Is this a classification or regression problem?
> It is a classification, since we are predicting if a customer will purchase again or not (yes or no)

5) How many features?
> There are 20 features

In [ ]:
# check shape
df1.shape

(5630, 20)

6) How many rows of data?
> There are 5,630 rows

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?
> From the header alone, I've noticed a few missing values already. Moreover, there are some features which may be unnecessary or irrelevant such as CustomerID

# Dataset 2: Customer Lifetime Value (CLTV)

1) Source of the data: Kaggle
[Customer Acquisition Data](https://www.kaggle.com/datasets/amirmotefaker/customer-acquisition-data)

2) Brief Description of data
> The data includes characteristics of customers. Specifically, it includes their customer acqusition cos, conversion rate and revenue.

In [ ]:
file2 = '/content/drive/MyDrive/Colab Notebooks/Data /customer_acquisition_data.csv'

df2=pd.read_csv(file2)
df2.head()

customer_id           channel       cost  conversion_rate  revenue
0            1          referral   8.320327         0.123145     4199
1            2  paid advertising  30.450327         0.016341     3410
2            3   email marketing   5.246263         0.043822     3164
3            4      social media   9.546326         0.167592     1520
4            5          referral   8.320327         0.123145     2419

3) What is the target?
> The target is to predict customer lifetime value (CLTV) or the estimated amount a customer will spend on a business.

4) Is this a classification or regression problem?
> It is a regression problem, since we are predicting the customer lifetime value

5) How many features?
> There are 5 features

In [ ]:
# check shape
df2.shape

(800, 5)

6) How many rows of data?
> There are 800 rows

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?
> With only 5 features and 800 rows, it could be that the size of the data is too small. However, this could be work if I use KNN or decision trees.

# Dataset 3: Predicting Startup Success

1) Source of the data: Kaggle
[StartUp Investments (Crunchbase)](https://www.kaggle.com/datasets/arindam235/startup-investments-crunchbase)

2) Brief Description of data
> The data includes characteristics of startup companies and investments. The data was collected by Crunchbase, a company that collects data about companies from its concepcion to Fortune 1000.

In [ ]:
file2 = '/content/drive/MyDrive/Colab Notebooks/Data /investments_VC.csv'

df3=pd.read_csv (file2, encoding='ISO-8859-1')
df3.head()

permalink                name  \
0            /organization/waywire            #waywire   
1  /organization/tv-communications  &TV Communications   
2    /organization/rock-your-paper   'Rock' Your Paper   
3   /organization/in-touch-network   (In)Touch Network   
4   /organization/r-ranch-and-mine  -R- Ranch and Mine   

                    homepage_url  \
0         http://www.waywire.com   
1          http://enjoyandtv.com   
2   http://www.rockyourpaper.org   
3  http://www.InTouchNetwork.com   
4                            NaN   

                                       category_list        market   \
0         |Entertainment|Politics|Social Media|News|          News    
1                                            |Games|         Games    
2                             |Publishing|Education|    Publishing    
3  |Electronics|Guides|Coffee|Restaurants|Music|i...   Electronics    
4                      |Tourism|Entertainment|Games|       Tourism    

   funding_total_usd      status country_code state_code         region  ...  \
0          17,50,000    acquired          USA         NY  New York City  ...   
1          40,00,000   operating          USA         CA    Los Angeles  ...   
2             40,000   operating          EST        NaN        Tallinn  ...   
3          15,00,000   operating          GBR        NaN         London  ...   
4             60,000   operating          USA         TX         Dallas  ...   

  secondary_market  product_crowdfunding round_A round_B round_C  round_D  \
0              0.0                   0.0     0.0     0.0     0.0      0.0   
1              0.0                   0.0     0.0     0.0     0.0      0.0   
2              0.0                   0.0     0.0     0.0     0.0      0.0   
3              0.0                   0.0     0.0     0.0     0.0      0.0   
4              0.0                   0.0     0.0     0.0     0.0      0.0   

  round_E round_F  round_G  round_H  
0     0.0     0.0      0.0      0.0  
1     0.0     0.0      0.0      0.0  
2     0.0     0.0      0.0      0.0  
3     0.0     0.0      0.0      0.0  
4     0.0     0.0      0.0      0.0  

[5 rows x 39 columns]

In [ ]:
# Try various encodings
encodings = ['utf-8', 'ISO-8859-1', 'latin1', 'Windows-1252', 'cp1252']

for encoding in encodings:
    try:
        df = pd.read_csv(file2, encoding=encoding)
        break  # Stop trying encodings if successful
    except UnicodeDecodeError:
        pass

df3.head()

permalink                name  \
0            /organization/waywire            #waywire   
1  /organization/tv-communications  &TV Communications   
2    /organization/rock-your-paper   'Rock' Your Paper   
3   /organization/in-touch-network   (In)Touch Network   
4   /organization/r-ranch-and-mine  -R- Ranch and Mine   

                    homepage_url  \
0         http://www.waywire.com   
1          http://enjoyandtv.com   
2   http://www.rockyourpaper.org   
3  http://www.InTouchNetwork.com   
4                            NaN   

                                       category_list        market   \
0         |Entertainment|Politics|Social Media|News|          News    
1                                            |Games|         Games    
2                             |Publishing|Education|    Publishing    
3  |Electronics|Guides|Coffee|Restaurants|Music|i...   Electronics    
4                      |Tourism|Entertainment|Games|       Tourism    

   funding_total_usd      status country_code state_code         region  ...  \
0          17,50,000    acquired          USA         NY  New York City  ...   
1          40,00,000   operating          USA         CA    Los Angeles  ...   
2             40,000   operating          EST        NaN        Tallinn  ...   
3          15,00,000   operating          GBR        NaN         London  ...   
4             60,000   operating          USA         TX         Dallas  ...   

  secondary_market  product_crowdfunding round_A round_B round_C  round_D  \
0              0.0                   0.0     0.0     0.0     0.0      0.0   
1              0.0                   0.0     0.0     0.0     0.0      0.0   
2              0.0                   0.0     0.0     0.0     0.0      0.0   
3              0.0                   0.0     0.0     0.0     0.0      0.0   
4              0.0                   0.0     0.0     0.0     0.0      0.0   

  round_E round_F  round_G  round_H  
0     0.0     0.0      0.0      0.0  
1     0.0     0.0      0.0      0.0  
2     0.0     0.0      0.0      0.0  
3     0.0     0.0      0.0      0.0  
4     0.0     0.0      0.0      0.0  

[5 rows x 39 columns]

In [ ]:
df3.columns

Index(['permalink', 'name', 'homepage_url', 'category_list', ' market ',
       ' funding_total_usd ', 'status', 'country_code', 'state_code', 'region',
       'city', 'funding_rounds', 'founded_at', 'founded_month',
       'founded_quarter', 'founded_year', 'first_funding_at',
       'last_funding_at', 'seed', 'venture', 'equity_crowdfunding',
       'undisclosed', 'convertible_note', 'debt_financing', 'angel', 'grant',
       'private_equity', 'post_ipo_equity', 'post_ipo_debt',
       'secondary_market', 'product_crowdfunding', 'round_A', 'round_B',
       'round_C', 'round_D', 'round_E', 'round_F', 'round_G', 'round_H'],
      dtype='object')

3) What is the target?
> The target is to predict the success of the Startup, where the goal is to determine which startup success and to determine which features leads to success. Success is determined the status of the company - operating / closed / acquired

4) Is this a classification or regression problem?
> It is a classification problem, since we are predicting the success of startups based on their status - operating / closed / acquired.

5) How many features?
> There are 39 features

In [ ]:
# check shape
df3.shape

(54294, 39)

6) How many rows of data?
> There are 54,294 rows

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?
> There are a few irrelevant columns such as permalink, name and homepage_url. There are many rows and features. Furthermore, I might need to check with features have correlation to Status, so I can remove those with no correlation to decrease the number of features.  